aiven/resource_project.go

```diff
@@ -12,21 +12,24 @@ import (
 
 var aivenProjectSchema = map[string]*schema.Schema{
 	"billing_address": {
-		Type:        schema.TypeString,
-		Description: "Billing name and address of the project",
-		Optional:    true,
+		Type:             schema.TypeString,
+		Description:      "Billing name and address of the project",
+		Optional:         true,
+		DiffSuppressFunc: emptyObjectDiffSuppressFunc,
 	},
 	"billing_emails": {
-		Type:        schema.TypeSet,
-		Description: "Billing contact emails of the project",
-		Elem:        &schema.Schema{Type: schema.TypeString},
-		Optional:    true,
+		Type:             schema.TypeSet,
+		Description:      "Billing contact emails of the project",
+		Elem:             &schema.Schema{Type: schema.TypeString},
+		Optional:         true,
+		DiffSuppressFunc: emptyObjectDiffSuppressFunc,
 	},
 	"billing_extra_text": {
-		Type:         schema.TypeString,
-		Description:  "Extra text to be included in all project invoices, e.g. purchase order or cost center number",
-		ValidateFunc: validation.StringLenBetween(0, 1000),
-		Optional:     true,
+		Type:             schema.TypeString,
+		Description:      "Extra text to be included in all project invoices, e.g. purchase order or cost center number",
+		ValidateFunc:     validation.StringLenBetween(0, 1000),
+		Optional:         true,
+		DiffSuppressFunc: emptyObjectDiffSuppressFunc,
```

Comment:
This sounds like something that should be possible to set to empty value. Possibly the same is true for some other fields as well. Won't this now prevent that?

Reply:
Yes, it is possible to set it to an empty string or ignore this field, for Terraform client it is all the same. The idea of adding this `DiffSuppressFunc` is to avoid situations when the user created a project using `copy_from_project`, which can populate some non zero value to this field and others. And we want to omit all these changes from a plan in case field is not set by the user to a particular value.

Reply:
Just to clarify: does this now work so that if the value is not present at all it will not be updated but if the value is present but empty it will be updated?

Reply:
Ah ok, now I get your point, in our case if a value is present, but empty TF client will ignore this change because `emptyObjectDiffSuppressFunc` will make it do so. We can have a different `DiffSuppressFunc` which will be a little bit smarter about that and will comply with the statement from above (`emptyObjectNoChangeDiffSuppressFunc`). 

Still, I believe in this case we would need to update go-client code because https://github.com/aiven/aiven-go-client/blob/master/project.go#L50 `UpdateProjectRequest ` has string pointers only for these fields: `billing_extra_text`, `billing_address` and `country_code` and the rest of optional fields will ignore by the go-client even if Terraform sets and empty string to the request.

WDYT, should we update a go-client or fields like `account_id`, `billing_currency`, `vat_id` should not have empty values?

Reply:
`billing_currency` cannot be empty. `vat_id` could be clearing it out is very rare. Probably same for `account_id`.

Reply:
Thanks, to clarify the situation completely, what about these fields:
- `card_id`
- `default_cloud`

All can be empty right?

And `copy_from_project` can only be set during the creation of resource?

Reply:
Yes to all